cmd/malwareExecuteScan_test.go

```diff
@@ -0,0 +1,169 @@
+
+func TestScanTests(t *testing.T) {
+
+	open = openFileMock()
+
+	getSHA256 = func(path string) (string, error) {
+		return "96ca802fbd54d31903f1115a1d95590c685160637d9262bd340ab30d0f817e85", nil
+	}
+	defer func() {
+		open = _open
+		getSHA256 = _getSHA256
+	}()
+
+	t.Run("No malware, no encrypted content", func(t *testing.T) {
+		httpClient := httpMock{StatusCode: 200, ResponseBody: "{\"malwareDetected\":false,\"encryptedContentDetected\":false,\"scanSize\":298782,\"mimeType\":\"application/octet-stream\",\"SHA256\":\"96ca802fbd54d31903f1115a1d95590c685160637d9262bd340ab30d0f817e85\"}"}
+
+		error := runMalwareScan(&malwareScanConfig, nil, nil, &httpClient)
+
+		if assert.NoError(t, error) {
+
+			t.Run("check url", func(t *testing.T) {
+				assert.Equal(t, "https://example.org/malwarescanner/scan", httpClient.URL)
+			})
+
+			t.Run("check method", func(t *testing.T) {
+				assert.Equal(t, "POST", httpClient.Method)
+			})
+
+			t.Run("check user", func(t *testing.T) {
+				assert.Equal(t, "me", httpClient.Options.Username)
+			})
+
+			t.Run("check password", func(t *testing.T) {
+				assert.Equal(t, "********", httpClient.Options.Password)
+			})
+		}
+	})
+
+	t.Run("Malware detected", func(t *testing.T) {
+
+		httpClient := httpMock{StatusCode: 200, ResponseBody: "{\"malwareDetected\":true,\"encryptedContentDetected\":false,\"scanSize\":298782,\"mimeType\":\"application/octet-stream\",\"SHA256\":\"96ca802fbd54d31903f1115a1d95590c685160637d9262bd340ab30d0f817e85\"}"}
+		error := runMalwareScan(&malwareScanConfig, nil, nil, &httpClient)
+		assert.EqualError(t, error, "Malware scan failed for file 'target/myFile'. Malware detected: true, encrypted content detected: false")
+	})
+
+	t.Run("Encrypted content detected", func(t *testing.T) {
+
+		httpClient := httpMock{StatusCode: 200, ResponseBody: "{\"malwareDetected\": false, \"encryptedContentDetected\": true, \"scanSize\":298782,\"mimeType\":\"application/octet-stream\",\"SHA256\":\"96ca802fbd54d31903f1115a1d95590c685160637d9262bd340ab30d0f817e85\"}"}
+		error := runMalwareScan(&malwareScanConfig, nil, nil, &httpClient)
+		assert.EqualError(t, error, "Malware scan failed for file 'target/myFile'. Malware detected: false, encrypted content detected: true")
+	})
+
+	t.Run("Malware and encrypted content detected", func(t *testing.T) {
+
+		httpClient := httpMock{StatusCode: 200, ResponseBody: "{\"malwareDetected\": true, \"encryptedContentDetected\": true, \"scanSize\":298782,\"mimeType\":\"application/octet-stream\",\"SHA256\":\"96ca802fbd54d31903f1115a1d95590c685160637d9262bd340ab30d0f817e85\"}"}
+		error := runMalwareScan(&malwareScanConfig, nil, nil, &httpClient)
+		assert.EqualError(t, error, "Malware scan failed for file 'target/myFile'. Malware detected: true, encrypted content detected: true")
+	})
+}
+
+func TestFileNotFound(t *testing.T) {
+
+	httpClient := httpMock{StatusCode: 200}
+
+	open = func(path string) (io.ReadCloser, error) {
```

Comment:
... and what happens in case the file denoted by `path` exists by ground-fog? With that approach we can be sure how it behaves and we explicitly describe how we expect the open method should behave in that case. Basically I dis-like to rely on a real file system in cases like that.